### DISCUSSION TITLE
Compile error with androidx

### DISCUSSION THREAD

**Discussion by philou78** [NightscoutFoundation/xDrip#3255] [Q&A]
Hello,
I am using xDrip for 6 years and I usually compile xDrip with Android Studio.
Recently I got this error:
```
.../app/src/androidTest/java/com/eveningoutpost/dexdrip/HomeEspressoTest.java:10: error: package android.support.test does not exist
import android.support.test.InstrumentationRegistry;
                           ^
```
I solved this by changing all the lines
```
import [static] android.support.test.XXXX 
```
with
```
import [static] androidx.test.test.XXXX
```
Then I got: 
```
.../app/src/androidTest/java/com/eveningoutpost/dexdrip/HomeEspressoTest.java:11: error: cannot find symbol
import androidx.test.annotation.UiThreadTest;
                               ^
  symbol:   class UiThreadTest
  location: package androidx.test.annotation
```

For this, I added to app/build.gradle:
```
    androidTestImplementation 'androidx.test:rules:1.5.0'
```
by selecting the latest available version 1.5.0

Is this fix correct ?

Thanks in advance for your support

**Comment by Navid200** [comment]
Why do you not use the official release?
https://navid200.github.io/xDrip/docs/ReleaseNotes.html

**Reply by philou78** [reply]
The official release has a long vibrate pattern and the delta format shows -.4 mg/d instead of -0.4 mg/dl. I prefer to set a shorter vibration pattern and better delta format that suit me better.

**Reply by Navid200** [reply]
Follow this:
https://github.com/NightscoutFoundation/xDrip/blob/master/Documentation/Quick_Start.md

Does it help?

**Reply by philou78** [reply]
This guide is very useful.

Actually I was using the default _fastDebug_ build variant.
When using _fastRelease_, test files such as _HomeEspressoTest.java_ are no longer compiled.

May I suggest adding the recommended build variant on [Quick_Start.md](https://github.com/NightscoutFoundation/xDrip/blob/master/Documentation/Quick_Start.md)?
If _HomeEspressoTest.java_ is still used for debugging, it also needs to be fixed.

Thanks for your support

**Comment by Navid200** [comment]
I have a question for you.
I am working on a submission to add the ability to choose one of different vibration patterns for each alert.
And I am going to include a set of patterns.  So, I need to choose them carefully to maximize the number of people who find them useful.

But, this has to be done in such a way to avoid causing problems.  Very long patterns can interfere with the ascending volume profile.
Very short patterns may never be played or may never be felt.

I measured the existing pattern and it only lasts 4 seconds.  You say it is long.  What do you mean by that?
Why is 4 seconds too long?
I am not being critical.  I want to understand the different preferences and the reason for them if possible.

Thanks

**Reply by philou78** [reply]
The xDrip alarms are a big help to me.
During working hours, in my open space, I prefer an alarm that looks like an SMS notification so as not to be noticed.
My modified vibrate pattern is equal to medium vibrate pattern of PR #2142
It's enough to wake me up at night

**Reply by Navid200** [reply]
1.3 seconds
  Ouch!

Well, if it wakes you up, that justifies compiling your own xDrip.

Thanks
